README.md

```diff
@@ -1,5 +1,79 @@
-# oauth-client
+# Promact-OAuth-Client
 
 [![Build Status](https://travis-ci.org/Promact/oauth-client.svg?branch=dev)](https://travis-ci.org/Promact/oauth-client)
 
 Client library for Promact OAuth server
+
+# .NET Client for Promact API
+
+This library allows you to interact with Promact OAuth Server endpoints in your .NET applications.
+It is fully asynchronous, designed to be non-blocking and object-oriented way to interact with Promact OAuth Server programmatically.
+
+# Installation
+You can add this library to your project using Nuget.This is the only method this library is currently distributed unless you choose to build your own binaries using source code. Run the following command in the �Package Manager Console�:
+```
+Install-Package OAuth.Client.Promact
+```
+Or right click to your project in Visual Studio, choose �Manage NuGet Packages� and search for **�OAuth.Client.Promact�** and click �Install�.
+# Usage
+You can initialize the middleware like the following :
+startup.cs
+```
+using Promact.OAuth.Client.Middleware;
+app.UsePromactAuthentication(new PromactAuthenticationOptions()
+{
+    Authority = "http://www.example.com/",
+    ClientId = "HJXF74EQ497GRAL",
+    ClientSecret = "ChcBmvtaUwphIsbmRkbL9amOh9Qy6Q",
+    LogoutUrl = "http://www.examples.com/",
+    AllowedScopes =
+    {
+        Scopes.offline_access,
+        Scopes.email,
+        Scopes.profile,
+        Scopes.openid,
+        Scopes.project_read,
+        Scopes.user_read,
+    },
+    Event = new OpenIdConnectEvents()
+    {
+        OnRemoteFailure = token => 
+        {
+            var userClaims = token.HttpContext.User.Claims;
+            return Task.FromResult(0);
+        }
+    }
+})
+```
+You can initialize the client like the following:
+```
+using Promact.OAuth.Client.Repository.User;
+using Promact.OAuth.Client.Repository.Project;
+public class Something
+{
+    private readonly IUserModule _user;
+    Private readonly IProjectModule _project;
+    public Something(IUserModule user, IProjectModule project)
```

Comment:
This is not a clear usage of the library. Are you assuming that anyone who is using will use it through Autofac or any other DI mechanism? 

Also, There is no instruction on how to initialize a module